packages/open-truss/src/uqi/uqi.ts

```diff
@@ -8,6 +8,78 @@ export type UqiMappedType =
   | 'Date'
   | 'JSON'
 
+// Parses and reformats a SynchronousQueryResult and
+// casts the values to the appropriate javascript type
+// SynchronousQueryResult:
+// {
+//   rows: [
+//     {
+//       values: {
+//         key: 'name', type: 'String', value: 'value'
+//       }
+//     }
+//   ]
+// }
+//
+// Result:
+// [
+//   { name: 'value' }
+// ]
+export function parseUqiResult(
+  result: SynchronousQueryResult,
+): Array<Record<string, unknown>> {
+  return result.rows.map((row) => {
+    const rowResult: Record<string, unknown> = {}
+    row.values.forEach(({ key, type, value }) => {
+      rowResult[key] = castUqiValue(type, value)
+    })
+    return rowResult
+  })
+}
+
+export function castUqiValue(type: string, value: string): unknown {
```

Comment:
Could we move this up next to `UqiMappedType`'s definition (line 3)? And could `type` be typed as `UqiMappedType` here?

```suggestion
export function castUqiValue(type: UqiMappedType, value: string): unknown {
```

Reply:
2e166dca9b366be9f57ab90d732b6b4662b99853